src/server/pkg/storage/fileset/util.go

```diff
@@ -82,3 +83,12 @@ func IsCleanTarPath(x string, isDir bool) bool {
 	y := CleanTarPath(x, isDir)
 	return y == x
 }
+
+func getSortedKeys(set map[string]struct{}) []string {
+	var keys []string
```

Comment:
nit: I like to pre-allocate with make here since you can get the since of the map.